simcon_project/users/views/researcher_home.py

```diff
@@ -10,11 +25,28 @@ def is_researcher(user):
 
 @user_passes_test(is_researcher)
 def researcher_view(request):
-    response_table = TemplateResponse.objects.all()
-    if request.method == "POST":
-        items_to_delete = request.POST.getlist('delete_items')
-        response_table.filter(pk__in=items_to_delete).delete()
-    paginator = Paginator(response_table, 10)
-    page_number = request.GET.get('page')
-    page_object = paginator.get_page(page_number)
-    return render(request, 'researcher_view.html', locals())
+    responses = TemplateResponse.objects.filter(
+        template__researcher__email=request.user)
+    filtered_responses = filter_search(request, responses)
+
+    if filtered_responses:
+        response_table = ResponseTable(filtered_responses)
+        RequestConfig(request, paginate={"per_page": 5}).configure(
+            response_table)
+    else:
+        response_table = None
+
+    context = {'responseTable': response_table}
+
+    return render(request, 'researcher_view.html', context)
+
+
+def filter_search(request, responses):
+    if 'searchParam' in request.GET:
+        param = request.GET['searchParam']
+        filter_fields = Q(student__first_name__contains=param) | Q(student__last_name__contains=param) | \
```

Comment:
I don't know how easy it would be to implement but searching by a student's full name filters out all the responses.